### PR TITLE
cmd/snap: close refresh notifications after trying to run a snap while inhibited

### PR DESCRIPTION
**Pull request by stolowski**
Close pending refresh notification from snap run after inhibiting snap execution due to ongoing refresh.

### CODE REVIEW

File: cmd/snap/cmd_run_test.go [hunk 1]
```diff
@@ -1931,6 +1941,34 @@ func (s *RunSuite) TestWaitWhileInhibitedGraphicalSessionFlowError(c *check.C) {
...
+func (s *RunSuite) TestWaitWhileInhibitedGraphicalSessionFlowErrorOnFinish(c *check.C) {
+	restoreIsGraphicalSession := snaprun.MockIsGraphicalSession(true)
+	defer restoreIsGraphicalSession()
+
+	restorePendingRefreshNotification := snaprun.MockPendingRefreshNotification(func(refreshInfo *usersessionclient.PendingSnapRefreshInfo) error {
+		return nil
+	})
+	defer restorePendingRefreshNotification()
+
+	restoreFinishRefreshNotification := snaprun.MockFinishRefreshNotification(func(refreshInfo *usersessionclient.FinishedSnapRefreshInfo) error {
```
Comment (bboozzoo) [inline cmd/snap/cmd_run_test.go:1953]: Is it possible to mock some calls deeper in the stack eg in usersessionclient?

Comment (mardy) [reply cmd/snap/cmd_run_test.go:1953]: I think that the current approach is the best one for unit testing. Testing the actual behaviour is better done in component tests (which we don't have) or integration tests (spread).

**Review by mardy — APPROVED** [review]
LGTM!

File: cmd/snap/cmd_run_test.go (hunk 1 above)
Comment (bboozzoo) [reply cmd/snap/cmd_run_test.go:1953]: I was hoping for something more elaborate, but it's clearly consistent with other unit tests in this package, so I guess it's fine.

**Review by bboozzoo — APPROVED** [review]
Thanks

**Pull request merged**